thorlcr/activities/indexwrite/thindexwriteslave.cpp

```diff
@@ -533,6 +535,8 @@ class IndexWriteSlaveActivity : public ProcessSlaveActivity, public ILookAheadSt
                             {
                                 CNodeInfo &info = tlkRows.item(idx);
                                 builder->processKeyData((char *)info.value, info.pos, info.size);
+                                if (info.size > maxRecordSizeSeen)
```

Comment:
Empirically by looking at indexes, we currently (before this change) pass key size 32767 when building the TLK for indexes that have variable records, and that is what is stored in length

Reply:
I'm not 100% sure if it needs resetting, but it's probably best to do so. I will add a line setting to zero at the point the builder is created.

Reply:
You would think that TLK should not have any payload in it, but it would seem to me from debugging operation of dumpKey on a TLK that it does so. I think there may already be a Jira around to remove it (perhaps I will look at that next)

Reply:
It further appears that the TLK rows also potentially contain blob IDs (that are invalid if any code should decide to try to resolve them)

Reply:
info.size will all be the same for the tlk, as only fixed length fields.
( you get an error if trying to include var length index fields when building )

It's using the max row size of the index in general, that's passed into the builder ctor, which for var length is KEYBUILD_MAXLENGTH by default. 

It's because it's reusing the builder instance for building the TLK.
So an alternate fix is to not reuse the builder instance and create a new one for TLK.
That may be slightly more correct anyway

Reply:
Re. the TLK containing payload and/or blob index ids ..
I see the reason, Thor is serializingthe last rows in their entirety (so a bit wasteful too), and passing those with lenghts to the TLK processKeyData.
It should be changed to serialize just the key fields.
I've opened a separate JIRA: https://track.hpccsystems.com/browse/HPCC-28628

Reply:
I think there was already a jira. I have a fix and am testing it to see if it creates compatibility issues...

Reply:
yep, Gavin's marked mine as a duplicate (of HPCC-24713)